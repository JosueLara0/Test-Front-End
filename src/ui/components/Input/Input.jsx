//* libraries
import { TextField } from "@mui/material";
import PropTypes from "prop-types";
//* styles
import { inputStyles as styles } from "./input.styles";

export const Input = ({ label, type, name, onChange }) => {
  return (
    <TextField
      label={label}
      type={type}
      name={name}
      sx={styles.textField}
      InputLabelProps={{
        style: {
          font: "normal normal bold 12px/14px Lato, sans serif",
          letterSpacing: "0px",
          opacity: 1,
          color: "#585858",
        },
      }}
      onChange={onChange}
    />
  );
};

//? Props validations
Input.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
