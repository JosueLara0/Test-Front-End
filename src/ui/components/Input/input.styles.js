export const inputStyles = {
    textField: {
        width: { xs: '80vw', md: '280px' },
        background: '#FFFFFF 0% 0% no-repeat padding-box',
        borderRadius: '5px',
        zIndex: 0
    }
};
