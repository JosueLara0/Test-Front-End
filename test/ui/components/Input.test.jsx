//* libraries
import { render } from "@testing-library/react";
//* components
import { Input } from "../../../src/ui/components/Input/Input";

describe("Tests in <Input />", () => {
  const label = "Ingresa tu nombre";
  const type = "text";
  const name = "name";
  const onInputChange = jest.fn();

  test("should make match with the snapshot", () => {
    const { container } = render(
      <Input label={label} type={type} name={name} onChange={onInputChange} />
    );

    expect(container).toMatchSnapshot();
  });
});
