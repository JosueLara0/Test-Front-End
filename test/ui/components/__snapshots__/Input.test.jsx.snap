// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Input /> should make match with the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root css-nuxnpp-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
      style="font: serif normal normal bold normal; letter-spacing: 0px; opacity: 1; color: rgb(88, 88, 88);"
    >
      Ingresa tu nombre
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        name="name"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1ftyaf0"
        >
          <span>
            Ingresa tu nombre
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
